builtInFunctions/esdtNFTCreate.go

```diff
@@ -275,6 +309,64 @@ func getLatestNonce(acnt vmcommon.UserAccountHandler, tokenID []byte) (uint64, e
 	return big.NewInt(0).SetBytes(nonceData).Uint64(), nil
 }
 
+func (e *esdtNFTCreate) getESDTNFTCreateInput(
+	vmInput *vmcommon.ContractCallInput,
+	tokenID []byte,
+	originalURIs [][]byte,
+	accountWithRoles vmcommon.UserAccountHandler,
+) (*esdtNFTCreateInput, error) {
+	args := vmInput.Arguments
+
+	var uris = originalURIs
+	var nonce uint64
+	var originalCreator []byte
+	var err error
+
+	isCrossChainToken := e.crossChainTokenCheckerHandler.IsCrossChainOperation(tokenID)
+	if !isCrossChainToken {
+		nonce, err = getLatestNonce(accountWithRoles, tokenID)
+		if err != nil {
+			return nil, err
+		}
+
+		originalCreator = vmInput.CallerAddr
+	} else {
+		nonce, originalCreator, err = getCrossChainTokenNonceAndCreator(args, vmInput.CallType)
+		if err != nil {
+			return nil, err
+		}
+
+		uris = uris[:len(uris)-2]
+	}
+
+	return &esdtNFTCreateInput{
+		nonce:                 nonce,
+		originalCreator:       originalCreator,
+		uris:                  uris,
+		isCrossChainOperation: isCrossChainToken,
+	}, nil
+}
+
+func getCrossChainTokenNonceAndCreator(args [][]byte, callType vm.CallType) (uint64, []byte, error) {
+	minRequiredArgs := minNumOfArgsForCrossChainMint
+	if callType == vm.ExecOnDestByCaller {
+		minRequiredArgs++
+	}
+
+	if len(args) < minRequiredArgs {
+		return 0, nil, fmt.Errorf("%w for cross chain token mint, last 2 arguments should be the nonce and original creator", ErrInvalidNumberOfArguments)
```

Comment:
If the callType is ExecOnDestByCaller, then the comment is not really true...
Maybe add another if at line 353.
Also please print the `len(args)` in the message, might be important some day.

Reply:
Refactored